test/test-parse-function.cpp

```diff
@@ -204,6 +204,40 @@ TEST_F(test_parse_function, async_function_statement) {
   }
 }
 
+TEST_F(test_parse_function, async_keyword_order_diagnostic1) {
+  {
+    test_parser p(u8"export async function f() {};"_sv);
+    p.parse_and_visit_statement();
+    ASSERT_EQ(p.variable_declarations.size(), 1);
+    EXPECT_EQ(p.variable_declarations[0].name, u8"f");
+  }
+}
+
+TEST_F(test_parse_function, async_keyword_order_diagnostic2) {
+  {
+    test_parser p(u8"async export function f() {};"_sv, capture_diags);
```

Comment:
Must fix: To ensure that the function was interpreted as `async`, we should write `await myPromise` in the body and check `p.variable_uses`. See the `await_in_async_function` in this file for an example.